Sources/IR/Analysis/ControlFlowGraph.swift

```diff
@@ -107,6 +107,57 @@ struct ControlFlowGraph: Sendable {
     return result
   }
 
+  /// The type of action to perform after exploring a vertex.
+  enum ExplorationContinuation {
+
+    /// Continue exploring the successors of the vertex.
+    case `continue`
+
+    /// Skip exploring the successors of the vertex.
+    case skip
+
+    /// Stop exploring the graph.
+    case stop
+
+  }
+
+  /// Explores the graph in a breadth-first manner starting from `start`, calling `handlingBlock`
+  /// for each visited block.
+  ///
+  /// When calling `handlingBlock`, the block's ID and its successors are passed as arguments. The
+  /// return value of `handlingBlock` determines how the exploration continues.
+  ///
+  /// If `forward` is `false`, the graph is explored in the reverse direction, following
+  /// predecessor edges instead of successor edges.
+  func exploreFrom(
+    _ start: [Block.ID],
+    forward: Bool = true,
+    handlingBlock: (Block.ID, [Block.ID]) -> ExplorationContinuation
```

Comment:
```suggestion
  /// Explores the graph in a breadth-first manner starting from `start`, calling `handleBlock`
  /// for each visited block.
  ///
  /// When calling `handlingBlock`, the block's ID and its successors are passed as arguments. The
  /// return value of `handlingBlock` determines how the exploration continues.
  ///
  /// If `forward` is `false`, the graph is explored in the reverse direction, following
  /// predecessor edges instead of successor edges.
  func exploreFrom(
    _ start: [Block.ID],
    forward: Bool = true,
    _ handleBlock: (Block.ID, [Block.ID]) -> ExplorationContinuation
```

Most of the time we use trailing-closure syntax in Swift so the label is typically irrelevant. If I had to put a label I would write `handlingBlockWith`.

Inside the function the name of the parameter should sounds more imperative I think.